gui/src/components/StoryComponents/StoryDetails.tsx

```diff
@@ -134,6 +139,39 @@ export default function StoryDetails({
         setStoryOverview(data.story.overview);
         setStoryTestCases(data.story.test_cases);
         setStoryInstructions(data.story.instructions);
+
+        if (data.story.status === 'IN_REVIEW') {
+          let issueTitle = '';
+          let issueDescription = '';
+          const actions = [];
+
+          switch (data.story.reason) {
+            case 'MAX_LOOP_ITERATION_REACHED':
+              issueTitle = 'Action Needed: Maximum number of iterations reached';
+              issueDescription = 'The story execution in the workbench has exceeded the maximum allowed iterations. You can update the story details and re-build it.';
+              actions.push(
+                  { label: 'Re-Build', link: 'REBUILD' },
+                  { label: 'Get Help', link: 'https://discord.com/invite/dXbRe5BHJC' }
+              );
+              break;
+            case 'IN_REVIEW_LLM_KEY_NOT_FOUND':
```

Comment:
move this to the constant folder and have a file there if one does not exist already